### PR TITLE
Add-Docker-Examples

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**
Add Docker Examples

### CODE REVIEW

File: docker/Ansible/ansible_container.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+              RUN curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
+              RUN python get-pip.py
+              RUN pip install --upgrade pip==9.0.1
+            - |-
+              RUN pip install ansible==
+            - { get_input: ansible_version }
+        tag:
+          concat:
+            - { get_input: image_tag }
+            - { get_input: ansible_version }
```
Comment (EarthmanT) [inline docker/Ansible/ansible_container.yaml:72]: How long does this take to build?

File: docker/Ansible/ansible_container.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+imports:
+  - http://cloudify.co/spec/cloudify/5.0.5/types.yaml
+  - plugin:cloudify-docker-plugin
+
+inputs:
+
+  docker_host:
+    type: string
+    default: ''
```
Comment (EarthmanT) [inline docker/Ansible/ansible_container.yaml:11]: Should we provide a default here? Maybe we should require the user to provide this IP?

File: docker/Ansible/ansible_container.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  docker_user:
+    type: string
+    default: centos
+
+  ansible_host:
+    type: string
+    default: 10.239.3.26
```
Comment (EarthmanT) [inline docker/Ansible/ansible_container.yaml:19]: Let's not provide default here.

File: docker/Ansible/ansible_container.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            - { get_input: ansible_version }
+
+  docker_ansible_container_files:
+    type: cloudify.nodes.docker.container_files
+    properties:
+      resource_config:
+        docker_machine:
+          docker_ip: { get_input: docker_host }
+          docker_user: { get_input: docker_user }
+          docker_key: { get_secret: agent_key_private }
```
Comment (EarthmanT) [inline docker/Ansible/ansible_container.yaml:81]: Can you use the key pair creation that @AdarShaked is adding to the other blueprints?

File: docker/Ansible/ansible_container.yaml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        source: https://github.com/ahmadiesa-abu/test-ansible-from-git.git
```
Comment (EarthmanT) [inline docker/Ansible/ansible_container.yaml:82]: Let's use a URL that's not in your personal account. Is there another good ansible that is publicly available from some "official" source?

File: docker/Ansible/ansible_container.yaml (hunk 2 above)
Comment (ahmadiesa-abu) [reply docker/Ansible/ansible_container.yaml]: removed the default , and made the required

File: docker/Ansible/ansible_container.yaml (hunk 4 above)
Comment (ahmadiesa-abu) [reply docker/Ansible/ansible_container.yaml]: the examples assume that you already have a machine configured, so you need to use the private key that was used for configuring the machine

File: docker/Ansible/ansible_container.yaml (hunk 5 above)
Comment (ahmadiesa-abu) [reply docker/Ansible/ansible_container.yaml]: used our example archive instead

File: docker/Ansible/ansible_container.yaml (hunk 3 above)
Comment (ahmadiesa-abu) [reply docker/Ansible/ansible_container.yaml]: done , made it required

File: docker/Ansible/ansible_container.yaml (hunk 1 above)
Comment (ahmadiesa-abu) [reply docker/Ansible/ansible_container.yaml]: it depends on docker host internet connection -but in general not long-

File: docker/terraform/terraform-container.yaml [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    type: string
+    default: { get_secret: aws_access_key_id }
+
+  aws_secret_access_key:
+    type: string
+    default: { get_secret: aws_secret_access_key }
+
+  aws_default_region:
+    type: string
+    default: { get_secret: ec2_region_name }
```
Comment (AdarShaked) [inline docker/terraform/terraform-container.yaml:52]: I think that in order to be consistent with our examples we should chage the input name to :"aws_region_name" with default value: "us-east-1"

File: docker/terraform/terraform-container.yaml [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+  public_key:
```
Comment (AdarShaked) [inline docker/terraform/terraform-container.yaml:54]: we can generate the key for terraform examples

**Pull request merged**